x/leverage/keeper/liquidate_test.go

```diff
@@ -203,3 +203,76 @@ func TestComputeLiquidation(t *testing.T) {
 	uDust.liquidationIncentive = sdk.MustNewDecFromStr("0")
 	runTestCase(uDust, 2, 1, 29, "high exchange rate collateral dust")
 }
+
+func TestCloseFactor(t *testing.T) {
+	require := require.New(t)
+
+	type testCase struct {
+		borrowedValue                sdk.Dec
+		collateralValue              sdk.Dec
+		liquidationThreshold         sdk.Dec
+		smallLiquidationSize         sdk.Dec
+		minimumCloseFactor           sdk.Dec
+		completeLiquidationThreshold sdk.Dec
+	}
+
+	baseCase := func() testCase {
+		return testCase{
+			sdk.MustNewDecFromStr("50"),  // borrowed value 50
+			sdk.MustNewDecFromStr("100"), // collateral value 100
+			sdk.MustNewDecFromStr("40"),  // liquidation threshold 40
+			sdk.MustNewDecFromStr("20"),  // small liquidation size 20
+			sdk.MustNewDecFromStr("0.1"), // minimum close factor 10%
+			sdk.MustNewDecFromStr("0.3"), // complete liquidation threshold 30%
+		}
+	}
+
+	runTestCase := func(tc testCase, expectedCloseFactor string, msg string) {
+		closeFactor := keeper.ComputeCloseFactor(
+			tc.borrowedValue,
+			tc.collateralValue,
+			tc.liquidationThreshold,
+			tc.smallLiquidationSize,
+			tc.minimumCloseFactor,
+			tc.completeLiquidationThreshold,
+		)
+
+		require.Equal(sdk.MustNewDecFromStr(expectedCloseFactor), closeFactor, msg)
+	}
+
+	// In the base case, close factor scales from 10% to 100% as borrowed value
+	// goes from liquidation threshold ($40) to a critical point, which is defined
+	// to be 30% of the way between liquidation threshold and collateral value ($100).
+	// This critical point is calculated to be borrowed value = $58. Since the actual
+	// borrowed value is $50, which is 5/9 the way from $40 to $58, the computed close
+	// factor is 5/9 of the way from 10% to 100% - thus, 60%.
+	runTestCase(baseCase(), "0.6", "base case")
+
+	// If borrowed value has passed the critical point, close factor is 1
+	completeLiquidation := baseCase()
+	completeLiquidation.borrowedValue = sdk.MustNewDecFromStr("60")
+	runTestCase(completeLiquidation, "1", "complete liquidation")
+
+	// If borrowed value is less than small liquidation size, close factor is always 1.
+	smallLiquidation := baseCase()
+	smallLiquidation.smallLiquidationSize = sdk.MustNewDecFromStr("60")
+	runTestCase(smallLiquidation, "1", "small liquidation")
+
+	// A liquidation-ineligible target would not have its close factor calculated in
+	// practice, but the function should return zero if it were.
+	notEligible := baseCase()
+	notEligible.borrowedValue = sdk.MustNewDecFromStr("40")
+	runTestCase(notEligible, "0", "liquidation ineligible")
```

Comment:
In this case we are at the threshold. I think we should already enable the liquidation, and the factor will be 10%.

Reply:
Changed, and added a test case for that explicitly